src/qobj/operators.jl

```diff
@@ -11,6 +12,52 @@ export commutator
 export tunneling
 export qft
 
+@doc raw"""
+    rand_unitary(dimensions, distribution=:haar)
+
+Returns a random unitary [`QuantumObject`](@ref).
+
+The `dimensions` can be either the following types:
+- `dimensions::Int`: Number of basis states in the Hilbert space.
+- `dimensions::Vector{Int}`: list of dimensions representing the each number of basis in the subsystems.
+
+The `distribution` specifies which of the method used to obtain the unitary matrix:
+- `:haar`: Haar random unitary matrix using the algorithm from reference 1
+- `:exp`: Uses ``\exp(-iH)``, where ``H`` is a randomly generated Hermitian operator.
+
+# References
+1. [F. Mezzadri, How to generate random matrices from the classical compact groups, arXiv:math-ph/0609050 (2007)](https://arxiv.org/abs/math-ph/0609050)
+"""
+rand_unitary(dimensions::Int, distribution::Symbol = :haar) = rand_unitary([dimensions], Val(distribution))
+rand_unitary(dimensions::Vector{Int}, distribution::Symbol = :haar) = rand_unitary(dimensions, Val(distribution))
+function rand_unitary(dimensions::Vector{Int}, ::Val{:haar})
```

Comment:
I have never seen changing a method using a Val of a symbol. It is ok, but maybe we can use some structs instead? In this way we leave the same consistency we changing the method of some process, like wigner, spectrum, steadystate and any other. What do you think?

Reply:
Actually, what I thought in the beginning is just using standard strings... (like `"haar"` or `"exp"` here), so that it can be the same as python qutip.
But we cannot use `Val{string}` to do multiple dispatch, so I decided to use `symbol`.

I think it will be a bit redundant to define structures and also the corresponding (object) constant for all these different methods. 

This might also cause troubles in the future. Because all the name of these methods would need to be exported. For example, exporting the method keyword `exp` here will cause conflicts with the `exp()` function. Then we need to give a new name for it, and that means the method name cannot be the same as python qutip.

Reply:
Okok, it is ok to me to do this approach. We need to keep the same syntax for all the package, so we have to change also the other methods for the other functions like spectrum, wigner, and so on..

Also, I was wondering about type inference for using this approach of `Val{:symbol}`. Could you add some type inference tests?

Reply:
I have added the inference tests.

I think it's not necessary to change the "solving methods", cause some solvers need extra arguments, it will be better to still use structure to save those arguments.